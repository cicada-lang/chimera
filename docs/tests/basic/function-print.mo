function f() {
  print "hi"
  let x = 1
  export let y = 2
  compute x

  function f(x, y) {
    return [x, y]
  }

  hyperrule abc {
    ["a", "b"] => ["c"]
    ["a", "c"] => ["b"]
    ["b", "c"] => ["a"]
  }

  print abc(quote ["a", "a", "a"])

  rule conjunctiveNormalForm {
    not(not(x)) => quote x
    not(and(x, y)) => quote or(not(x), not(y))
    not(or(x, y)) => quote and(not(x), not(y))
    or(and(x, y), z) => quote and(or(x, z), or(y, z))
    or(x, and(y, z)) => quote and(or(x, y), or(x, z))
  }

  print conjunctiveNormalForm(
    quote not(and(not(A), not(B)))
  )

  if false {
    print 0
  } else if false {
    print 1
  } else if false {
    print 2
  } else {
    print 3
  }

  return
}

print f
compute f()
