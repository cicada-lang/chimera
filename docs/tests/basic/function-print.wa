print () => {
  print "hi"
  let x = 1
  private let y = 2
  compute x
  function f(x, y) {
    return [x, y]
  }
  return
}
