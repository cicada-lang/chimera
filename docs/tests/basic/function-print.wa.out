() => {
  print "hi"
  let x = 1
  private let y = 2
  compute x
  function f(x, y) {
    return [x, y]
  }
  hyperrule abc {
    ["a", "b"] => ["c"]
    ["a", "c"] => ["b"]
    ["b", "c"] => ["a"]
  }
  print abc(quote ["a", "a", "a"])
  rule conjunctiveNormalForm {
    not(not(x)) => x
    not(and(x, y)) => or(not(x), not(y))
    not(or(x, y)) => and(not(x), not(y))
    or(and(x, y), z) => and(or(x, z), or(y, z))
    or(x, and(y, z)) => and(or(x, y), or(x, z))
  }
  print conjunctiveNormalForm(quote not(and(not(A), not(B))))
  
  return null
}
"hi"
["a", "a", "a"]
or(A, B)
3
